### DISCUSSION TITLE
A chart library *light* enough to run on embedded? (WildernessLabs Meadow)

### DISCUSSION THREAD

**Discussion by houstonhaynes** [Live-Charts/LiveCharts2#1205] [Ideas]
A few of us on the WildernessLabs Slack are wondering if there's a *truly* lightweight .NET charting library out there which can be supported on that micro-controller. This just released a "MicroLayout" feature for HMI devices, which supports partial screen rendering for those devices which support it. 

Any ideas? Thanks!

**Comment by beto-rodriguez** [comment]
The heavy part on LiveCharts, is Skia/SkiaSharp, if the micro controller can load Skia, then LiveCharts should not be a problem, I am afraid that I am not sure if we can load skia there.

But the good news is that the core of LiveCharts does not require Skia at all, LiveCharts has a simple framework that handles animations, updates, and all the cool features on the library, the core of LiveCharts is just a light math library, then we connect the SkiaSharp backed to render the images in the UI.

This sounds like an interesting feature, and I think we could make it work.

**Reply by houstonhaynes** [reply]
Thanks so much for echoing back. One of the core developers of the WildernessLabs Meadow Core platform said that the rendering library would have to be managed and honestly I'm not sure how/whether that relates to Skia (memory footprint? Runtime?). I suppose that would be a subject of some homework on my part. FWIW the in-house designs are centered on the STM32F7 with Nuttx RTOS and Mono on top of that. Everything runs on netstandard2.0 but I think they're migrating to full Core (6, maybe?)

**Comment by upswing1** [comment]
That would be great if it could run on Meadow.

**Comment by houstonhaynes** [comment]
I relayed this thread back to the folks building the core and graphics libraries for the WildernessLabs Meadow. Although their collective plates are full at the moment, they're putting a pin in their roadmap to see what's plausible and possible.
